packages/headless/src/features/insight-user-actions/insight-user-actions-loader.ts

```diff
@@ -0,0 +1,54 @@
+import {PayloadAction, AsyncThunkAction} from '@reduxjs/toolkit';
+import {AsyncThunkInsightOptions} from '../../api/service/insight/insight-api-client';
+import {CoreEngine} from '../../app/engine';
+import {insightUserActionsReducer as insightUserActions} from '../../features/insight-user-actions/insight-user-actions-slice';
+import {
+  registerUserActions,
+  fetchUserActions,
+  RegisterUserActionsPayload,
+  StateNeededByFetchUserActions,
+  FetchUserActionsThunkReturn,
+} from './insight-user-actions-actions';
+
+export type {RegisterUserActionsPayload};
+export interface InsightUserActionsActionCreators {
+  /**
+   * Registers the user actions.
+   *
+   * @param payload - The action creator payload.
+   * @returns A dispatchable action.
+   */
+  registerUserActions(
+    payload: RegisterUserActionsPayload
+  ): PayloadAction<RegisterUserActionsPayload>;
```

Comment:
What exactly is "registerUserActions" ?
The comment on what this is should be updated to tell you what you are registering :)